From: Ryan Lortie <desrt@desrt.ca>
Date: Wed, 20 Nov 2013 21:57:26 +0000
Subject: valgrind.h: add "r0" to the clobber list on PPC

Looks like the magic sequences trash this register, so make sure GCC
knows that.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=710983
Origin: upstream, 2.39.2, commit:6c8600b2b86fe42edd6d6e9648a92f32e404457c
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=737379
---
diff --git a/glib/valgrind.h b/glib/valgrind.h
index cc2cf3d..7bb1382 100644
--- a/glib/valgrind.h
+++ b/glib/valgrind.h
@@ -442,7 +442,7 @@ typedef
                      "mr %0,3"     /*result*/                     \
                      : "=b" (_zzq_result)                         \
                      : "b" (_zzq_default), "b" (_zzq_ptr)         \
-                     : "cc", "memory", "r3", "r4");               \
+                     : "cc", "memory", "r3", "r4", "r0");         \
     _zzq_result;                                                  \
     })
 
@@ -455,7 +455,7 @@ typedef
                      "mr %0,3"                                    \
                      : "=b" (__addr)                              \
                      :                                            \
-                     : "cc", "memory", "r3"                       \
+                     : "cc", "memory", "r3", "r0"                 \
                     );                                            \
     _zzq_orig->nraddr = __addr;                                   \
   }
@@ -504,7 +504,7 @@ typedef
                      "mr %0,3"     /*result*/                     \
                      : "=b" (_zzq_result)                         \
                      : "b" (_zzq_default), "b" (_zzq_ptr)         \
-                     : "cc", "memory", "r3", "r4");               \
+                     : "cc", "memory", "r3", "r4", "r0");         \
     _zzq_result;                                                  \
   })
 
@@ -517,7 +517,7 @@ typedef
                      "mr %0,3"                                    \
                      : "=b" (__addr)                              \
                      :                                            \
-                     : "cc", "memory", "r3"                       \
+                     : "cc", "memory", "r3", "r0"                 \
                     );                                            \
     _zzq_orig->nraddr = __addr;                                   \
     __asm__ volatile(__SPECIAL_INSTRUCTION_PREAMBLE               \
@@ -526,7 +526,7 @@ typedef
                      "mr %0,3"                                    \
                      : "=b" (__addr)                              \
                      :                                            \
-                     : "cc", "memory", "r3"                       \
+                     : "cc", "memory", "r3", "r0"                 \
                     );                                            \
     _zzq_orig->r2 = __addr;                                       \
   }
--
cgit v0.9.2
